src/arrayMethodSort.js

```diff
@@ -4,8 +4,19 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const comparator =
+      compareFunction || ((a, b) => (String(a) > String(b) ? 1 : -1));
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (comparator(this[j], this[j + 1]) > 0) {
+          [this[j], this[j + 1]] = [this[j + 1], this[j]];
+        }
+      }
+    }
```

Comment:
The bubble sort implementation is correct, but it's worth noting that bubble sort is not the most efficient sorting algorithm. Consider optimizing it or using a more efficient algorithm if performance is a concern.